sys/debug/kgmon.py

```diff
@@ -0,0 +1,106 @@
+import gdb
+import sys
+
+
+from ctypes import *
+from .cmd import SimpleCommand
+from .struct import GdbStructMeta
+
+PROGRES_BAR_SIZE = 20
+
+
+def nextBar(nomin, denom, bar):
+    return int(nomin * bar / denom) < int((nomin + 1) * bar / denom)
+
+
+class RawArc(Structure):
+    _fields_ = [('raw_frompc', c_uint), ('raw_selfpc', c_uint),
+                ('raw_count', c_int)]
+
+
+class ToStruct(Structure):
+    _fields_ = [('selfpc', c_uint), ('count', c_int),
```

Comment:
`selfpc` is `u_long`, `count` is `long` in `gmon.h`. Is it ok?

Reply:
It is ok, at this moment KPROF works for mips and we need just a 32 bit type, using `c_ulong` use 64 bits.